src/torchpme/calculators/calculator.py

```diff
@@ -215,6 +219,16 @@ def _validate_compute_parameters(
                 f"({device})"
             )
 
+        determinant = torch.linalg.det(cell)
+
+        if smearing is not None and torch.isclose(
+            determinant, torch.tensor(0.0, dtype=cell.dtype, device=cell.device)
+        ):
```

Comment:
I think we want to check if it is exactly zero, right? If the cell is super small it should be fine.

```suggestion
        if smearing is not None and torch.equal(
            cell.det(), torch.tensor(0.0, dtype=cell.dtype, device=cell.device)
        ):
```